### DISCUSSION TITLE
Update the code with a tree action

### DISCUSSION THREAD

**Discussion by codeDude64** [nvim-tree/nvim-tree.lua#1336] [Q&A] [answered]
Is it possible update the code if the tree made a update on the tree directory?

For example if I have something like
```
src
|
--coolCode
  |
  --main.js
  --main.css
  --anotherFile.jpg
```
And I rename one of them for example main.js. Is it possible that the code where main.js is imported be updated to the new name?

or if I change the position of a file in the directory tree something like
```
src
|
--anotherFile.jpg
--coolCode
  |
  --main.js
  --main.css
```
Is it possible to update the code where the image is imported?

**Comment by alex-courtis** [comment, marked as answer]
That's well outside the scope of nvim-tree.

You might be able to use an [LSP plugin](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md) to achieve this functionality: refactor / rename the usage of anotherFile.jpg in your code and have it move the file.

**Reply by kyazdani42** [reply]
That might be outside of the scope of nvim-tree but i was thinking adding functionality for that, either in a separate plugin or in the repository (simple LSP integration).
For now, use your LSP client to do that by binding to nvim-tree event system `:help nvim-tree.events`

**Reply by EricDriussi** [reply]
That would be really cool, I'm trying to DIY something like that myself but I cant seem to make it work.

Any idea where to look for instructions to achieve something like this? Specifically the LSP side, the nvim-tree events seem pretty self-explanatory :smile:

**Reply by alex-courtis** [reply]
> Any idea where to look for instructions to achieve something like this? Specifically the LSP side, the nvim-tree events seem pretty self-explanatory

`:help lsp` describes everything LSP but at a very low level. Not a great place to start.

[nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) provides a higher level framework to work within. Ideas:
* try to find an existing configuration that meets your requirements [server_configurations](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md)
* extend an existing configuration
* write your own configuration

**Comment by peaske7** [comment]
@alex-courtis 
You may have already solved this yourself, but I found https://github.com/antosha417/nvim-lsp-file-operations to work well.

**Reply by alex-courtis** [reply]
That is an incredible plugin, thanks for highlighting in.

Someone has added a wiki entry: https://github.com/nvim-tree/nvim-tree.lua/wiki/Integrations-And-Extension-Plugins#nvim-lsp-file-operations
